### PR TITLE
Feature/create schema

### PR DESCRIPTION
**Pull request by joshLong145**
- Functionality to create `Schemas` which persist to storage mechanisms

## Changes
- Addition of the following proto definitions
    - WhatIs
    - SchemaReference
    - SchemaDefinition

## API Updates
- /v1/schema/create (HTTP Highway)
### Deprecations *(required)*
- None

## Checklist

- [x] Unit tests
- [ ] Documentation

## References *(optional)*

Fixes 
Connects 
closes #374 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: x/schema/keeper/msg_server_schema.go [hunk 1]
```diff
@@ -3,13 +3,69 @@ package keeper
...
+	}
+
+	var schema = types.SchemaReference{
+		Label: msg.Label,
+		Did:   what_is_did.String(),
+		Cid:   msg.Cid,
+	}
+
+	var whatIs = types.WhatIs{
+		Creator:   creator_did.GetID().String(),
```
Comment (mcjcloud) [inline x/schema/keeper/msg_server_schema.go]: Seems like you're creating a did document in `creator_did` only to take the DID string from it (which you already had)

File: x/schema/keeper/schema.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	"encoding/binary"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/google/uuid"
+	"github.com/sonr-io/sonr/x/schema/types"
+)
+
+func (k Keeper) GenerateKeyForDID() string {
+	id := uuid.New()
```
Comment (mcjcloud) [inline x/schema/keeper/schema.go]: nit: `return uuid.New().String()`

File: x/schema/keeper/schema.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+// GetSchemaFromCreator returns a WhoIs whos DIDDocument contains the given controller
+func (k Keeper) GetWhatIsFromCreator(ctx sdk.Context, creator string) (val []types.WhatIs, found bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.SchemaKeyPrefix))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+	defer iterator.Close()
+	var vals []types.WhatIs = make([]types.WhatIs, 0)
+	for ; iterator.Valid(); iterator.Next() {
+		var instance types.WhatIs
+		k.cdc.MustUnmarshal(iterator.Value(), &instance)
```
Comment (mcjcloud) [inline x/schema/keeper/schema.go]: I don't think we should panic if this fails. We won't be able to return a valid error response

File: x/schema/module.go [hunk 4]
```diff
@@ -77,7 +79,7 @@ func (AppModuleBasic) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Rout
 
 // RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the module.
 func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
-	// this line is used by starport scaffolding # 2
```
Comment (mcjcloud) [inline x/schema/module.go:80]: Don't remove this comment

**Review by mcjcloud — APPROVED** [review]

File: x/schema/keeper/msg_server_schema.go (hunk 1 above)
Comment (prdlk) [reply x/schema/keeper/msg_server_schema.go]: Yes better to just create a method in `pkg/did` which parses any arbitrary snrAddress and converts it to the `did:snr:` equivalent

File: x/schema/keeper/schema.go (hunk 2 above)
Comment (prdlk) [reply x/schema/keeper/schema.go]: @ntindle Thoughts?

**Review by prdlk — DISMISSED** [review]

File: x/schema/keeper/msg_server_schema.go (hunk 1 above)
Comment (joshLong145) [reply x/schema/keeper/msg_server_schema.go]: good point, i'll make that change.

**Review by prdlk — APPROVED** [review]

**Pull request merged**